src/mapper/pkg/clouduploader/cloud_upload.go

```diff
@@ -207,13 +207,18 @@ func (c *CloudUploader) NotifyAWSIntents(ctx context.Context, intents []awsinten
 	err := c.client.ReportDiscoveredIntents(
 		ctx,
 		lo.Map(intents, func(intent awsintentsholder.AWSIntent, _ int) *cloudclient.DiscoveredIntentInput {
+			var awsRole *string
+			if intents[0].IamRole != "" {
```

Comment:
why `intents[0]` and not `intent`? fix or explain in comment cause this is confusing